lnrpc/walletrpc/walletkit_util_test.go

```diff
@@ -72,3 +74,57 @@ func TestParseDerivationPath(t *testing.T) {
 		})
 	}
 }
+
+// TestGetWitnessType tests that the mapping between the witness type enum
```

Comment:
Nice progress in this PR 👍 
I feel kind of unsure about this test, whether that's the right way we wanna approach this. Lets see what more experienced coders say here. 

What do you think about using a golang map in the getWitnessType function `map[input.WitnessType] WitnessType` and in the test looping through it and comparing them with the `WitnessType_name` map? Then we would catch both directions of diverging types?

Reply:
It sounds like @ziggie1984's suggestion is similar to what I've done [here](https://github.com/lightningnetwork/lnd/pull/6903/commits/3a8aa575188b48c8b2d1f2c22874b76e5fbf8cb2#diff-97a234788862027c261c89226cdc9a681e96f8803888e57cdddfef9aad229da1R4440). Basically list the expected types manually and then compare it to the proto defs?

Reply:
I hear what you guys are saying. The test is a bit weird the way it is right now.

I think using a map for `getWitnessType` and also basing the test off that map makes a lot of sense.

The only downside is that someone could still add a new witness type to `input.StandardWitnessType`, but forget to add it to the mapping, and the test won't pick that up. But I guess if it's not being added to the mapping, then it couldn't really be used in any case.

Let me rework the PR- this is the plan: I will add a `map[input.WitnessType]WitnessType` to `walletkit_server.go`, use that map in the `getWitnessType` function, and then in the test, iterate over this map and compare that to the proto definition. Does this make sense?

Reply:
I've updated the PR. 

I've done as discussed above, except I didn't create a `getWitnessType` function. The logic is only used in one place (`walletrpc.PendingSweeps`) so no need to extract into a function. I just used the `allWitnessTypes` mapping directly instead.